### PR TITLE
perf(Theme): add theme change animation

### PR DESCRIPTION
**Pull request by WitMiao**
Add theme change animation

Screenshot gif:
iShot_2024-01-04_15 08 07

> Use [View Transitions API](https://developer.mozilla.org/en-US/docs/Web/API/View_Transitions_API), Browser support:
image

### CODE REVIEW

**Review by Fine0830 — APPROVED** [review]
LGTM

File: src/layout/components/NavBar.vue [hunk 1]
```diff
@@ -119,6 +126,35 @@ limitations under the License. -->
...
+  function handleChangeTheme() {
+    const x = themeSwitchRef.value?.offsetLeft ?? 0;
+    const y = themeSwitchRef.value?.offsetTop ?? 0;
+    const endRadius = Math.hypot(Math.max(x, innerWidth - x), Math.max(y, innerHeight - y));
+    // compatibility handling
+    if (!document.startViewTransition) {
+      changeTheme();
+      return;
+    }
+    // api: https://developer.chrome.com/docs/web-platform/view-transitions
```
Comment (wu-sheng) [inline src/layout/components/NavBar.vue:138]: What does this comment mean?

Comment (Fine0830) [reply src/layout/components/NavBar.vue:138]: It seems to be an API introduction.

Comment (WitMiao) [reply src/layout/components/NavBar.vue:138]: It's the api introduction, not necessary, can delete it.

**Pull request merged**